etcdserver/api/v3rpc/key.go

```diff
@@ -134,9 +134,15 @@ func checkPutRequest(r *pb.PutRequest) error {
 	if len(r.Key) == 0 {
 		return rpctypes.ErrGRPCEmptyKey
 	}
+	if r.IgnoreLease && r.IgnoreValue {
```

Comment:
I do not think it matters. If users want just to update the revision, they can. And we do not even need to check (save code logic).

opinion? @heyitsanthony